renderer.php

```diff
@@ -105,10 +105,20 @@ public function render_envbar($match, $fixed = true, $envs = array()) {
         // Show the configured env message.
         $showtext = htmlspecialchars($match->showtext);
 
+        // If this is the env we are on we trust $config->prodlastcheck.
+        // Else we can trust $match->lastrefresh.
+        $lastrefresh = 0;
+        $here = (new moodle_url('/'))->out();
+        if (envbarlib::is_match($here, $match->matchpattern)) {
+            $config = get_config('local_envbar');
+            $lastrefresh = $config->prodlastcheck;
+        } else {
+            $lastrefresh = $match->lastrefresh;
+        }
+
         // Just show the biggest time unit instead of 2.
-        $config = get_config('local_envbar');
-        if (isset($config->prodlastcheck)) {
-            $show = format_time(time() - $config->prodlastcheck);
+        if ($lastrefresh > 0) {
+            $show = format_time(time() - $lastrefresh);
```

Comment:
Can you please move $config->prodlastcheck and $lastrefresh to be an attribute in $env like $env[0]->lastrefreshed so all this renderer logic simpler